src/System.ComponentModel.TypeConverter/src/System/Drawing/FontConverter.cs

```diff
@@ -0,0 +1,399 @@
+                    style |= FontStyle.Underline;
+            }
+
+            /* ?? Should default font be culture dependent ?? */
+            if (name == null)
+            {
+                fontFamily = new FontFamily("Tahoma");
+            }
+            else
+            {
+                FontCollection collection = new InstalledFontCollection();
+                FontFamily[] installedFontList = collection.Families;
+                foreach (FontFamily font in installedFontList)
+                {
+                    if (name.Equals(font.Name, StringComparison.OrdinalIgnoreCase))
+                    {
+                        fontFamily = font;
+                        break;
+                    }
+                }
+
+                // font family not found in installed fonts
+                if (fontFamily == null)
+                {
+                    collection = new PrivateFontCollection();
+                    FontFamily[] privateFontList = collection.Families;
+                    foreach (FontFamily font in privateFontList)
+                    {
+                        if (name.Equals(font.Name, StringComparison.OrdinalIgnoreCase))
+                        {
+                            fontFamily = font;
+                            break;
+                        }
+                    }
+                }
+
+                // font family not found in private fonts also
+                if (fontFamily == null)
+                    fontFamily = FontFamily.GenericSansSerif;
+            }
+
+            return new Font(fontFamily, size, style, unit, charSet, vertical);
+        }
+
+        public override bool GetCreateInstanceSupported(ITypeDescriptorContext context) => true;
+
+        public override PropertyDescriptorCollection GetProperties(
+            ITypeDescriptorContext context,
+            object value,
+            Attribute[] attributes)
+        {
+            return value is Font ? TypeDescriptor.GetProperties(value, attributes) : base.GetProperties(context, value, attributes);
+        }
+
+        public override bool GetPropertiesSupported(ITypeDescriptorContext context) => true;
+
+        public sealed class FontNameConverter : TypeConverter, IDisposable
+        {
+            FontFamily[] _fonts;
+
+            public FontNameConverter()
+            {
+                _fonts = FontFamily.Families;
+            }
+
+            void IDisposable.Dispose()
+            {
+            }
+
+            public override bool CanConvertFrom(ITypeDescriptorContext context, Type sourceType)
+            {
+                return sourceType == typeof(string) ? true : base.CanConvertFrom(context, sourceType);
+            }
+
+            public override object ConvertFrom(ITypeDescriptorContext context, CultureInfo culture, object value)
+            {
+                return value is string ? MatchFontName((string)value, context) : base.ConvertFrom(context, culture, value);
```

Comment:
you can use `value is string` to declare inline variable and no need to cast it again.